calicoctl/commands/node/status.go

```diff
@@ -0,0 +1,203 @@
+	doc := `Usage:
+calicoctl node status
+
+Description:
+  Display the status of the Calico node`
+
+	_, _ = docopt.Parse(doc, args, true, "calicoctl", false, false)
+
+	ctx := context.Background()
+
+	defaultHeaders := map[string]string{"User-Agent": "engine-api-cli-1.0"}
+	cli, err := client.NewClient("unix:///var/run/docker.sock", "v1.22", nil, defaultHeaders)
+	if err != nil {
+		return err
+	}
+
+	options := types.ContainerListOptions{All: true}
+	containers, err := cli.ContainerList(ctx, options)
+	if err != nil {
+		panic(err)
+	}
+
+	for _, c := range containers {
+		if strings.Contains(c.Names[0], "calico-node") && c.State == "running" {
+
+			fmt.Printf("calico-node container is %s. Status: %s\n", c.State, c.Status)
+
+			// config := types.ExecConfig{
+			// 	User:         "root",
+			// 	AttachStdin:  true,
+			// 	AttachStderr: true,
+			// 	AttachStdout: true,
+			// 	//Detach:       true,
+			// 	Cmd: []string{"sh", "-c", "cat libraries.txt"},
+			// }
+
+			// exec, err := cli.ContainerExecCreate(context.Background(), c.ID, config)
+			// if err != nil {
+			// 	fmt.Printf("failed to create exec: %v", err)
+			// 	break
+			// }
+
+			// reply, err := cli.ContainerExecAttach(context.Background(), exec.ID, config)
+			// if err != nil {
+			// 	fmt.Printf("failed to attach to the container: %v", err)
+			// 	break
+			// }
+			// defer reply.Close()
+
+			// b := make([]byte, 255)
+
+			// _, err = reply.Reader.Read(b)
+			// if err != nil {
+			// 	fmt.Printf("failed to exececute command inside the container: %v", err)
+			// 	break
+			// }
+
+			// re, err := regexp.Compile(`calico\s(.*)\n`)
+			// if err != nil {
+			// 	fmt.Printf("failed to compile regex: %v", err)
+			// 	break
+			// }
+
+			// parts := strings.Fields(re.FindString(string(b)))
+			// fmt.Println(parts)
+
+			// if len(parts) == 2 {
+			// 	felixVersion := parts[1][1 : len(parts[1])-1]
+
+			// 	fmt.Printf("Running felix version %v\n", felixVersion)
+			// }
+
+			// Connect to the bird socket and get the data if calico-node container is running
+			c, err := net.Dial("unix", "/var/run/calico/bird.ctl")
+			if err != nil {
+				log.Fatalf("Error connecting to the BIRD socker: %v", err)
+			}
+			defer c.Close()
+
+			fmt.Println()
+
+			_, _ = c.Write([]byte("show protocols\n"))
+			if err != nil {
+				log.Fatal("Error writing to BIRD socket:", err)
+			}
+			printBIRDResponse(c)
+
+			return nil
+		}
+	}
+
+	// Return and print message if calico-node is not running
+	fmt.Println("calico-node container not running")
+	return nil
+
+}
+
+func printBIRDResponse(r io.Reader) {
+	buf := make([]byte, 1024)
+
+	n, err := r.Read(buf[:])
+	if err != nil {
+		return
+	}
+
+	resp := string(buf[:n])
+	drawTable(resp)
+}
+
+func drawTable(birdOut string) {
+	data := [][]string{}
+
+	ipSep := "."
+	if !ipv4 {
+		ipSep = ":"
+	}
+
+	table := tablewriter.NewWriter(os.Stdout)
+	table.SetHeader([]string{"Peer address", "Peer type", "State", "Since", "Info"})
```

Comment:
yeah, we're going to need to re-structure this code a bit to make IPv6 work.

It shouldn't be too crazy, but let's chat about it when you have a chance.  Given the alpha release closing in, I say we do that in a separate PR, but let's do it _soon_.